src/test_utils.rs

```diff
@@ -136,6 +136,41 @@ pub fn open_gpkg_for_update(path: &Path) -> (TempPath, Dataset) {
     (temp_path, ds)
 }
 
+/// Copies the given file to a temporary file and opens it for writing. When the returned
+/// `TempPath` is dropped, the file is deleted.
+pub fn open_dataset_for_update(path: &Path) -> (TempPath, Dataset) {
+    use std::fs;
+    use std::io::Write;
+
+    let input_data = fs::read(path).unwrap();
+    let (mut file, temp_path) = tempfile::Builder::new()
+        // using the whole filename as suffix should be fine (can't
+        // use .extension() for .shp.zip and such)
+        .suffix(
+            path.file_name()
+                .unwrap_or_default()
+                .to_string_lossy()
+                .as_ref(),
+        )
+        .tempfile()
+        .unwrap()
+        .into_parts();
+    file.write_all(&input_data).unwrap();
```

Comment:
I just copied the function `open_gpkg_for_update`. But I have modified based on what you said.

Irrelevant, but I was thinking of making the `copy_file` option for `tempfile::Builder` so that we can just provide a file, and it'll make a temporary file that is a copy of the file provided using `std::fs::copy`. That would save a lot of lines of code, but it seems to be external package, maybe a pull request idea.